src/Nethermind/Nethermind.JsonRpc/Data/AccessListItemForRpc.cs

```diff
@@ -17,33 +16,36 @@ public class AccessListItemForRpc
         public AccessListItemForRpc(Address address, IReadOnlyCollection<UInt256>? storageKeys)
         {
             Address = address;
-            StorageKeys = storageKeys?.ToArray() ?? Array.Empty<UInt256>();
+            StorageKeys = storageKeys?.ToList() ?? new List<UInt256>();
         }
 
         public Address Address { get; set; }
 
         [JsonProperty(ItemConverterType = typeof(StorageCellIndexConverter))]
-        public UInt256[]? StorageKeys { get; set; }
+        public List<UInt256>? StorageKeys { get; set; }
 
-        public static AccessListItemForRpc[] FromAccessList(AccessList accessList) =>
-            accessList.Data.Select(kvp => new AccessListItemForRpc(kvp.Key, kvp.Value)).ToArray();
+        public static AccessListItemForRpc[] FromAccessList(AccessList accessList)
+        {
+            return accessList.AsEnumerable()
+                .Select(tuple => new AccessListItemForRpc(tuple.Address, tuple.StorageKeys.ToList()))
+                .ToArray();
```

Comment:
potentially, can it be left as IEnumerable's? It will be only serialized, so could avoid a bit of allocations?